SPECS/cifs-utils/CVE-2020-14342.patch

```diff
@@ -0,0 +1,121 @@
+diff -Naur cifs-utils-6.8.orig/mount.cifs.c cifs-utils-6.8.mod/mount.cifs.c
+--- cifs-utils-6.8.orig/mount.cifs.c	2020-09-30 17:26:48.250924409 -0700
++++ cifs-utils-6.8.mod/mount.cifs.c	2020-09-30 17:27:19.002733900 -0700
+@@ -1646,6 +1646,73 @@
+ 	return 0;
+ }
+ 
++#ifdef ENABLE_SYSTEMD
++static int get_passwd_by_systemd(const char *prompt, char *input, int capacity)
++{
++	int fd[2];
++	pid_t pid;
++	int offs = 0;
++	int rc = 1;
++
++	if (pipe(fd) == -1) {
++		fprintf(stderr, "Failed to create pipe: %s\n", strerror(errno));
++		return 1;
++	}
++
++	pid = fork();
++	if (pid == -1) {
++		fprintf(stderr, "Unable to fork: %s\n", strerror(errno));
++		close(fd[0]);
++		close(fd[1]);
++		return 1;
++	}
++	if (pid == 0) {
++		close(fd[0]);
++		dup2(fd[1], STDOUT_FILENO);
++		if (execlp("systemd-ask-password", "systemd-ask-password", prompt, NULL) == -1) {
++			fprintf(stderr, "Failed to execute systemd-ask-password: %s\n",
++			strerror(errno));
++		}
++		exit(1);
++	}
++
++	close(fd[1]);
++	for (;;) {
++		if (offs+1 >= capacity) {
++			fprintf(stderr, "Password too long.\n");
++			kill(pid, SIGTERM);
++			rc = 1;
++			break;
++		}
++		rc = read(fd[0], input + offs, capacity - offs);
++		if (rc == -1) {
```

Comment:
This should check if `errno == EINTR` and retry the loop with `continue` rather than failing right here.